### ISSUE TITLE
MongoDB transaction results are ignored

### ISSUE THREAD

**Issue by klaussner** [meteor/meteor#10195]
Transactions have their own oplog entries, which aren't handled by Meteor's `OplogObserveDriver`. If a transaction is committed successfully, the driver will receive an oplog entry like the one below, where `op == "c"` and `applyOps` contains the operations performed in the transaction.

**Expected behavior**

Operations in `applyOps` are handled like normal operations.

**Actual behavior**

Transaction entries are ignored and no data is sent to clients. This is because the oplog selector doesn't include
```
{ op: 'c', 'o.applyOps': { $exists: true } }
```
and doesn't consider entries in the `admin.$cmd` namespace (`ns`):
https://github.com/meteor/meteor/blob/543d4d3d51ba8c85fa697544e5f67174458c07c9/packages/mongo/oplog_tailing.js#L39-L47

**Reproduction**

https://github.com/klaussner/meteor-issues/tree/10195. If you run `Meteor.call("transaction")` in the browser console, the method will update two documents with random values but the client won't receive the new data.

**Meteor version**

1.7.1(-rc.5)

**Example oplog entry**
```json
{
  "ts":"6597672440761942017",
  "t":"1",
  "h":"6774736952743729336",
  "v":2,
  "op":"c",
  "ns":"admin.$cmd",
  "wall":"2018-09-05T09:38:41.196Z",
  "lsid":{
    "id":"rxMCGypxSJ2IfKxI5LJDOQ==",
    "uid":"47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU="
  },
  "txnNumber":"2",
  "stmtId":0,
  "prevOpTime":{
    "ts":"0",
    "t":"-1"
  },
  "o":{
    "applyOps":[
      {
        "op":"u",
        "ns":"meteor.collection",
        "ui":"0TYxDak8SumhJ1SwL6RAbw==",
        "o":{
          "$v":1,
          "$set":{
            "anotherField":0.5870170165354542
          }
        },
        "o2":{
          "_id":"A"
        }
      },
      {
        "op":"u",
        "ns":"meteor.collection",
        "ui":"0TYxDak8SumhJ1SwL6RAbw==",
        "o":{
          "$v":1,
          "$set":{
            "anotherField":0.2196245396309442
          }
        },
        "o2":{
          "_id":"B"
        }
      }
    ]
  }
}
```

**Comment by benjamn** [comment]
It seems like we need this to be able to claim transaction support in Meteor 1.7.1 with MongoDB 4.0, right?

**Comment by mitar** [comment]
I think that it might be also useful if we could pass `session` to existing Meteor MongoDB operations inside options argument. Same for [arrayFilters](https://github.com/meteor/meteor-feature-requests/issues/268#issuecomment-418178202) to claim support for MongoDB 3.6. :-)

**Comment by klaussner** [comment]
@benjamn Yeah, this should definitely be supported. I will have a PR ready soon! 🏃💨

**Comment by klaussner** [comment]
No problem, I'm un-assigning myself. :sweat_smile:

**Comment by benjamn** [comment]
This specific issue should be fixed if you update to the latest Meteor 1.7.1 release candidate:
```bash
meteor update --release 1.7.1-rc.8
```
Thanks to @klaussner for identifying the necessary changes! And don't worry @mitar, we'll get to your other suggestions soon.
